packages/devtools_app/lib/main.dart

```diff
@@ -38,33 +38,57 @@ void main() {
     const DevToolsScreen(screen: InspectorScreen()),
     DevToolsScreen(
       screen: const TimelineScreen(),
-      controllerProvider: (child) => TimelineControllerProvider(child: child),
+      controllerProvider: (child) => ControllerProvider<TimelineController>(
```

Comment:
you can replace the controllerProvider line with 
createController: (_) =>TimelineController()
and then in the implementation of DevToolsScreen create a Provider<TController> that passes in the createController function as the `create` argument.